runtime/src/main/java/org/corfudb/runtime/CorfuRuntime.java

```diff
@@ -879,46 +873,32 @@ private void pruneRemovedRouters(@Nonnull Layout layout) {
             int systemDownTriggerCounter = 0;
 
             while (true) {
-
-                Collections.shuffle(layoutServersCopy);
-                // Iterate through the layout servers, attempting to connect to one
-                for (String s : layoutServersCopy) {
-                    log.debug("Trying connection to layout server {}", s);
-                    try {
-                        IClientRouter router = getRouter(s);
-                        // Try to get a layout.
-                        CompletableFuture<Layout> layoutFuture =
-                                new LayoutClient(router, Layout.INVALID_EPOCH).getLayout();
-                        // Wait for layout
-                        Layout l = layoutFuture.get();
-
-                        // If the layout we got has a smaller epoch than the latestLayout epoch,
-                        // we discard it.
-                        if (latestLayout != null && latestLayout.getEpoch() > l.getEpoch()) {
-                            log.warn("fetchLayout: Received a layout with epoch {} from server "
-                                            + "{}:{} smaller than latestLayout epoch {}, "
-                                            + "discarded.",
-                                    l.getEpoch(), router.getHost(), router.getPort(),
-                                    latestLayout.getEpoch());
-                            continue;
-                        }
-
-                        checkClusterId(l);
-
-                        layout = layoutFuture;
-                        latestLayout = l;
-                        log.debug("Layout server {} responded with layout {}", s, l);
-
-                        // Prune away removed node routers from the nodeRouterPool.
-                        pruneRemovedRouters(l);
-
-                        return l;
-                    } catch (InterruptedException ie) {
-                        throw new UnrecoverableCorfuInterruptedError(
-                                "Interrupted during layout fetch", ie);
-                    } catch (Exception e) {
-                        log.warn("Tried to get layout from {} but failed with exception:", s, e);
+                try {
+                    // Fetch the latest layout we can possibly fetch from the list of layout servers.
+                    CompletableFuture<Layout> layoutFuture = getLatestLayout(layoutServersCopy);
```

Comment:
Why is the runtime tracking two layouts (i.e. layout and latestLayout), remove latestLayout and just update the this.layout as you discover layouts with higher epochs.